packages/http-crawler/src/internals/http-crawler.ts

```diff
@@ -0,0 +1,845 @@
+     * ```
+     * // Will fall back to windows-1250 encoding if none found
+     * suggestResponseEncoding: 'windows-1250'
+     * ```
+     */
+    suggestResponseEncoding?: string;
+
+    /**
+     * By default this crawler will extract correct encoding from the HTTP response headers. Use `forceResponseEncoding`
+     * to force a certain encoding, disregarding the response headers.
+     * To only provide a default for missing encodings, use {@apilink HttpCrawlerOptions.suggestResponseEncoding}
+     * ```
+     * // Will force windows-1250 encoding even if headers say otherwise
+     * forceResponseEncoding: 'windows-1250'
+     * ```
+     */
+    forceResponseEncoding?: string;
+
+    /**
+     * Automatically saves cookies to Session. Works only if Session Pool is used.
+     *
+     * It parses cookie from response "set-cookie" header saves or updates cookies for session and once the session is used for next request.
+     * It passes the "Cookie" header to the request with the session cookies.
+     */
+    persistCookiesPerSession?: boolean;
+}
+
+/**
+ * @internal
+ */
+export type InternalHttpHook<Context> = (
+    crawlingContext: Context,
+    gotOptions: OptionsInit,
+) => Awaitable<void>;
+
+export type HttpHook<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = Dictionary,
+> = InternalHttpHook<HttpCrawlingContext<UserData, JSONData>>;
+
+/**
+ * @internal
+ */
+export interface InternalHttpCrawlingContext<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = Dictionary,
+    Crawler = HttpCrawler<any>
+    > extends CrawlingContext<UserData> {
+    /**
+     * The request body of the web page.
+     * The type depends on the `Content-Type` header of the web page:
+     * - String for `text/html`, `application/xhtml+xml`, `application/xml` MIME content types
+     * - Buffer for others MIME content types
+     */
+    body: (string | Buffer);
+
+    /**
+     * The parsed object from JSON string if the response contains the content type application/json.
+     */
+    json: JSONData;
+
+    /**
+     * Parsed `Content-Type header: { type, encoding }`.
+     */
+    contentType: { type: string; encoding: BufferEncoding };
+    crawler: Crawler;
+    response: IncomingMessage;
+    sendRequest: (overrideOptions?: Partial<GotOptionsInit>) => Promise<GotResponse<string>>;
+}
+
+export interface HttpCrawlingContext<UserData = any, JSONData = any>
+    extends InternalHttpCrawlingContext<UserData, JSONData, HttpCrawler<HttpCrawlingContext<UserData, JSONData>>> {}
+
+export type HttpRequestHandler<
+    UserData extends Dictionary = any, // with default to Dictionary we cant use a typed router in untyped crawler
+    JSONData extends Dictionary = Dictionary,
```

Comment:
This should probably be `= any` instead (JSON data can be anything after all)

Reply:
I copied this from the cheerio crawler